frontend/src/e2e-playwright/browser.ts

```diff
@@ -21,23 +32,42 @@ beforeAll(async () => {
   browser = await playwright[config.playwright.browser].launch({
     headless: config.playwright.headless
   })
+  global.evaka = { takeScreenshots }
 })
 
-afterEach(async () => {
+beforeEach(async () => {
   for (const ctx of browser?.contexts() ?? []) {
     await ctx.close()
   }
 })
 
 afterAll(async () => {
   await browser?.close()
+  delete global.evaka
 })
 
 const injected = fs.readFileSync(
   path.resolve(__dirname, '../e2e-test-common/injected.js'),
   'utf-8'
 )
 
+async function takeScreenshots(namePrefix: string): Promise<void> {
+  if (!browser) return
+  const pages = browser.contexts().flatMap((ctx, ctxIndex) =>
```

Comment:
`pages.length` seems to always be 0 at this point. However, changing the event handler to listen to `test_fn_failure` seems to do the trick.

Reply:
Yeah it looks like some but not all of our tests do `afterEach(() => page?.close())` which means the page is closed before the `test_done` event. Those `page.close()` calls are unnecessary so we could remove them, but maybe `test_fn_failure` is in any case a better event for this use case.